pkgs/jni/lib/src/third_party/global_env_extensions.dart

```diff
@@ -1411,155 +1446,132 @@ class GlobalJniEnv {
       _NewDirectByteBuffer(address, capacity).objectPointer;
 
   late final _GetDirectBufferAddress = ptr.ref.GetDirectBufferAddress
-      .asFunction<JniPointerResult Function(JObjectPtr buf)>();
+      .asFunction<JniPointerResult Function(JObjectPtr buf)>(isLeaf: true);
 
   ffi.Pointer<ffi.Void> GetDirectBufferAddress(JObjectPtr buf) =>
       _GetDirectBufferAddress(buf).getPointer<ffi.Void>();
 
   late final _GetDirectBufferCapacity = ptr.ref.GetDirectBufferCapacity
-      .asFunction<JniResult Function(JObjectPtr buf)>();
+      .asFunction<JniResult Function(JObjectPtr buf)>(isLeaf: true);
 
   int GetDirectBufferCapacity(JObjectPtr buf) =>
       _GetDirectBufferCapacity(buf).long;
 
-  late final _GetObjectRefType =
-      ptr.ref.GetObjectRefType.asFunction<JniResult Function(JObjectPtr obj)>();
+  late final _GetObjectRefType = ptr.ref.GetObjectRefType
+      .asFunction<JniResult Function(JObjectPtr obj)>(isLeaf: true);
 
   int GetObjectRefType(JObjectPtr obj) => _GetObjectRefType(obj).integer;
-}
 
-/// Wraps over the function pointers in JniAccessorsStruct and exposes them as
```

Comment:
The accessor struct and JNI env are two different data structures in `jni.h`? Why? Can de be used interchangably? Does it mean that we're now lumping these together in our own API? If yes, then is `Jni.env` the right name for our combined API?

(Please add some more documentation on the PR for why we are making this change.)

Reply:
Added more doc to the PR.